src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs

```diff
@@ -280,6 +286,13 @@ private static void AddLinuxContainerServices(this IServiceCollection services)
                     var hostingConfigOptions = s.GetService<IOptions<FunctionsHostingConfigOptions>>();
                     return new LinuxContainerMetricsPublisher(environment, standbyOptions, logger, hostNameProvider, hostingConfigOptions);
                 }
+                else if (environment.IsLinuxConsumptionOnLegion())
```

Comment:
Need to determine the correct environment check to identify CV1 apps running on Legion. I believe @balag0 and @Tonewall were working on this.

As it stands, I don't think this check is entirely correct.